python/private/config_settings.bzl

```diff
@@ -101,6 +113,25 @@ def construct_config_settings(*, name, default_version, versions, minor_mapping)
             visibility = ["//visibility:public"],
         )
 
+    _current_config(
+        name = "current_config",
+        build_setting_default = "",
+        settings = documented_flags + [_PYTHON_VERSION_FLAG.name],
+        visibility = ["//visibility:private"],
+    )
+    native.config_setting(
+        name = "is_not_matching_current_config",
+        # We use the rule above instead of @platforms//:incompatible so that the
+        # printing of the current env always happens when the _current_config rule
+        # is executed.
+        #
+        # NOTE: This should in practise only happen if there is a missing compatible
+        # `whl_library` in the hub repo created by `pip.parse`.
+        flag_values = {"current_config": "will-never-match"},
```

Comment:
Are you sure this won't print spurious warnings?

IIRC, Bazel will evaluate all the select condition keys because it ensures that only one matches. This would imply that given e.g. `select({"py38": ..., ":current_config": ...})`, that `_print_current_config_impl()` is going to run, even if it won't match, and thus would print a message.

Reply:
So the whole structure is:
```
alias(
    name = "repo_alias",
    actual = select({
        "hub_select_condition_1": "repo_1",
        "hub_select_condition_2": "repo_2"
        "//conditions:default": "no_match_alias"
    }),
)

alias(
    name = "no_match_alias",
    select({"this-custom-incompatible-that-prints": "//python:none"}
    tags = ["manual"], # do we need this?
)
```

So my thinking is that the `no_match_alias` will not be analysed unless hub select conditions don't match, because the `no_match_alias` is not the key of the select dict but the value.

Unless we do `bazel build @pypi//...` We should not build the `no_match_alias` target. Maybe we have to have a manual tag?

Reply:
> So the structure is default maps to no-match-alias

Ahhh, ok. I was looking for this structure but missed it. LGTM.

> should we add manual to no-match-alias?

Yes. It just avoids a false failure when `...` et al accidentally pick it up. Which would probably be when poking around, debugging, or some situation where the false failure is probably going to be annoying.